cmd/plakar/subcommands/locate/locate.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2025 Gilles Chehade <gilles@poolp.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package locate
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/PlakarKorp/plakar/appcontext"
+	"github.com/PlakarKorp/plakar/cmd/plakar/subcommands"
+	"github.com/PlakarKorp/plakar/cmd/plakar/utils"
+	"github.com/PlakarKorp/plakar/objects"
+	"github.com/PlakarKorp/plakar/repository"
+	"github.com/PlakarKorp/plakar/snapshot"
+)
+
+func init() {
+	subcommands.Register("locate", cmd_locate)
+}
+
+func cmd_locate(ctx *appcontext.AppContext, repo *repository.Repository, args []string) (int, error) {
+	var opt_snapshot string
+
+	flags := flag.NewFlagSet("locate", flag.ExitOnError)
+	flags.StringVar(&opt_snapshot, "snapshot", "", "snapshot to locate in")
+	flags.Parse(args)
+
+	snapshotIDs := make([]objects.Checksum, 0)
+	if opt_snapshot != "" {
+		tmp := utils.LookupSnapshotByPrefix(repo, opt_snapshot)
+		snapshotIDs = append(snapshotIDs, tmp...)
+	} else {
+		tmp, err := repo.GetSnapshots()
+		if err != nil {
+			ctx.GetLogger().Error("%s: could not list snapshots: %s", flags.Name(), err)
+			return 1, err
+		}
+		snapshotIDs = append(snapshotIDs, tmp...)
+	}
```

Comment:
nit: we can simplify this down to:

```go
var snapshotIDs [][]objects.Checksum
if opt_snapshot != "" {
	snapshotIDs = utils.LookupSnapshotByPrefix(repo, opt_snapshot)
} else {
	var err error
	snapshotIDs, err = repo.GetSnapshots()
	if err != nil {
		ctx.GetLogger().Error("...")
		return &, err
	}
}
```

which has the effect that it doesn't make me wonder what are we adding to where.

Reply:
indeed, pushed as a new commit